fix: fix TransactionManagers ABI break

tpetracca opened the pull request:
In #6674 I accidentally introduced an ABI break when I deleted the `TransactionManagers#registrar` interface method.
This both:
- fixes the break by bringing the method back
- applies revapi to `atlasdb-config` to prevent this from happening again

==COMMIT_MSG==
fix TransactionManagers ABI break
==COMMIT_MSG==

pkoenig10 commented on atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:
```diff
@@ -169,6 +169,16 @@ public abstract class TransactionManagers {
+     * it would be an ABI break.
+     */
+    @Value.Default
+    @Deprecated
```
Might not want to deprecate this since it will block automatic upgrades in any repos that have `-Werror` and aren't actively maintained.

jeremyk-91 reviewed (approved):
👍  Argh, yeah I was thinking revapi would catch us if we got rid of anything. Let's not deprecate for the reasons @pkoenig10 notes?

(And now revapi is exploding because it thinks we're adding the registrar)

Putting DNM just to make sure we don't miss the above, but otherwise should be good to go.

tpetracca commented:
okay so I've
- removed the deprecation
- configured rev-api to use `0.909.0` as its comparison point so that we're instead "accepting the breaks we're keeping from #6674"
- reviewed and actually ensured each of the accepted breaks is in fact acceptable...

breaks:
- two constructors for `DefaultLockAndTimestampServiceFactory` -> this class is not instantiated anywhere outside of this library
- removed `Leaders` -> this class is not referenced in any code outside of this library
- removed `ImmutableLocalPaxosServices` and `ImmutableRemotePaxosServerSpec` -> these classes are not referenced in any code outside of this library, and neither are the interfaces they're generated from.

svc-autorelease commented:
Released [0.911.0](https://github.com/palantir/atlasdb/releases/tag/0.911.0)
